tools/ports/__init__.py

```diff
@@ -197,24 +197,24 @@ def fetch_project(name, url, subdir, sha512hash=None):
     if local_ports:
       logger.warning('using local ports: %s' % local_ports)
       local_ports = [pair.split('=', 1) for pair in local_ports.split(',')]
-      with shared.Cache.lock('local ports'):
-        for local_name, path in local_ports:
-          if name == local_name:
-            port = ports_by_name.get(name)
-            if not port:
-              utils.exit_with_error('%s is not a known port' % name)
-            if not hasattr(port, 'SUBDIR'):
-              utils.exit_with_error(f'port {name} lacks .SUBDIR attribute, which we need in order to override it locally, please update it')
-            subdir = port.SUBDIR
-            target = os.path.join(fullname, subdir)
-            if os.path.exists(target) and not dir_is_newer(path, target):
-              logger.warning(f'not grabbing local port: {name} from {path} to {fullname} (subdir: {subdir}) as the destination {target} is newer (run emcc --clear-ports if that is incorrect)')
-            else:
-              logger.warning(f'grabbing local port: {name} from {path} to {fullname} (subdir: {subdir})')
+      for local_name, path in local_ports:
+        if name == local_name:
+          port = ports_by_name.get(name)
+          if not port:
+            utils.exit_with_error('%s is not a known port' % name)
+          if not hasattr(port, 'SUBDIR'):
+            utils.exit_with_error(f'port {name} lacks .SUBDIR attribute, which we need in order to override it locally, please update it')
+          subdir = port.SUBDIR
+          target = os.path.join(fullname, subdir)
+          if os.path.exists(target) and not dir_is_newer(path, target):
+            logger.warning(f'not grabbing local port: {name} from {path} to {fullname} (subdir: {subdir}) as the destination {target} is newer (run emcc --clear-ports if that is incorrect)')
```

Comment:
We compare `path` and `target` - shouldn't we lock to avoid racing with another process?

Reply:
That would mean locking at least once for each invocation, and that would mean deadlocking whenever we run and emcc subprocess with lock held.   These days we assert that no subprocess will ever need to take the lock if its parent already holds it... since that is generally true since I refactored things in #17434.  

I'll see if I can use the same double-checked locking tick here..

Reply:
I used the same double-checked locking we down below.. should be good now.